Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,449 @@
+                        const int orderFee = 0;
+                        OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee, "Bitfinex Cancel Order Event") { Status = OrderStatus.Canceled });
+                    }
+                    else
+                    {
+                        return false;
+                    }
+
+                }
+            }
+            catch (Exception err)
+            {
+                Log.Error("CancelOrder(): OrderID: " + order.Id + " - " + err);
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Setup ticker polling
+        /// </summary>
+        public override void Connect()
+        {
+            _tickerToken = new CancellationTokenSource();
+        }
+
+        /// <summary>
+        /// Cancel ticker polling
+        /// </summary>
+        public override void Disconnect()
+        {
+            if (this._tickerToken != null)
+            {
+                this._tickerToken.Cancel();
+            }
+        }
+
+        private List<BitfinexOrder> GetOpenBitfinexOrders()
+        {
+            var list = new List<BitfinexOrder>();
+
+            try
+            {
+                var response = _restClient.GetActiveOrders();
+                if (response != null)
+                {
+                    foreach (var item in response)
+                    {
+                        list.Add(new BitfinexOrder
+                        {
+                            Quantity = Convert.ToInt32(decimal.Parse(item.OriginalAmount) * ScaleFactor),
+                            BrokerId = new List<string> { item.Id.ToString() },
+                            Symbol = Symbol,
+                            Time = Time.UnixTimeStampToDateTime(double.Parse(item.Timestamp)),
+                            Price = decimal.Parse(item.Price) / ScaleFactor,
+                            Status = MapOrderStatus(item),
+                            OriginalAmount = decimal.Parse(item.OriginalAmount) * ScaleFactor,
+                            RemainingAmount = decimal.Parse(item.RemainingAmount) * ScaleFactor,
+                            ExecutedAmount = decimal.Parse(item.ExecutedAmount) * ScaleFactor
+                        });
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive orders from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Order> GetOpenOrders()
+        {
+
+            var list = this.GetOpenBitfinexOrders().Select(o => (Order)o).ToList();
+
+            foreach (var item in list)
+            {
+                if (item.Status != OrderStatus.Canceled && item.Status != OrderStatus.Filled && item.Status != OrderStatus.Invalid)
```

Comment:
There's a handy [extension method](https://github.com/QuantConnect/Lean/blob/master/Common/Orders/OrderExtensions.cs#L40) to test if an order status is open.